cabal-install/Distribution/Client/Dependency.hs

```diff
@@ -178,6 +181,10 @@ data PackagePreference =
      -- | If we prefer versions of packages that are already installed.
    | PackageInstalledPreference PackageName InstalledPreference
 
+   | PackageStanzasPreference   PackageName [OptionalStanza]
```

Comment:
Obviously, this is some preference related to stanzas. But what kind of preference is it? Are we trying to respect the dependencies added by the optional stanzas? Needs a comment.